core/embed/rust/src/ui/model_tr/component/address_details.rs

```diff
@@ -0,0 +1,270 @@
+use heapless::Vec;
+
+use crate::{
+    error::Error,
+    micropython::buffer::StrBuffer,
```

Comment:
Please no StrBuffer under `ui/`, and micropython only in justified cases. Make the components polymorphic with `T: AsRef<str>`.

Reply:
Yes, it was a lot of changes and troubleshooting. Done in [a90c84c](https://github.com/trezor/trezor-firmware/pull/2610/commits/a90c84ce45bf5df551c0c12c624d13fd71251101)